test/system/runScenario3.js

```diff
@@ -0,0 +1,17 @@
+const runScenario3 = (trader2) => {
+  console.log('********************* Scenario 3 :  Test balance checks *********************');
+  return trader2.placeLimitOrder('S', trader2.getBalance().assets + 1, 0.14)
```

Comment:
After placing an order that is expected to fail, we need to test that the promise is rejected (because if the order is successfully placed without available balance, that's a serious bug)

We can use the .then()'s second argument:
```javascript
callSomeFunctionThatWillReject()
  .then(
    () => { throw new Error("The function resolved when it shouldn't have"),
    () => { /* the function rejected, this is expected and good */ }
  );
```

It would probably make sense to add an extra argument to each of the Trader's functions...
```javascript
placeLimitOrder(side, quantity, price, expectSuccess = true) {
  // ...
  if (expectSuccess) {
    this.createOrder(price, side, quantity).then(resolve, reject);
  } else {
    // we expect this call to fail
    this.createOrder(price, side, quantity)
      .then(
        () => { reject(new Error("createOrder resolved when it shouldn't have")),
        () => { resolve(); }
      );
  }
}
```